### PR TITLE
Equation solving

### LINKED ISSUE #244: Equation solving
As a user, I want to directly see the result of an equation I entered in order to minimize my interactions with Algernon.

### PR DESCRIPTION
**Pull request by matveysiva**
- Resolves #244

### CODE REVIEW

File: packages/Algernon.package/ALGExpressionType.class/instance/getDisplayedResult.st [hunk 1]
```diff
@@ -0,0 +1,11 @@
+accessing
+getDisplayedResult
+
+	|command equationsResult|
+	
+	command := self getCommand asString.
+	
+	self isEquation
+		ifTrue: [equationsResult := Compiler evaluate: command for: self logged: false.
```
Comment (yannik-dittmar) [inline packages/Algernon.package/ALGExpressionType.class/instance/getDisplayedResult.st]: Maybe a try catch here to prevent unwanted error messages from showing up and making Algernon unusable.

**Review by yannik-dittmar — CHANGES_REQUESTED** [review]
Syntax error when using `^`:
image

Why is the result down at the bottom when putting two spaces between the numbers? Maybe we could force the expression item to be at the top when an equation is detected?
image

Divide by zero error (solution idea stated in file comment):
image

**Comment by LinqLover** [comment]
> Syntax error when using `^`:
> image

FYIO, mathematical power in Smalltalk is expressed via `**`:

```smalltalk
2**3 "8"
```

**Comment by yannik-dittmar** [comment]
> > Syntax error when using `^`:
> > image
> 
> FYIO, mathematical power in Smalltalk is expressed via `**`:
> 
> ```smalltalk
> 2**3 "8"
> ```

Thank you for your info, but then the regex shouldn't match this as an equation.

File: packages/Algernon.package/ALGExpressionType.class/class/equationPattern.st [hunk 2]
```diff
@@ -1,4 +1,4 @@
 icon
 equationPattern
 
-	^ '( *[+\-]? *\d+)( *[+\-*\/^]? *\d+)* *'
\ No newline at end of file
+	^ '( *[+\-]? *\d+)( *[+\-*\/]? *\d+)* *'
```
Comment (yannik-dittmar) [inline packages/Algernon.package/ALGExpressionType.class/class/equationPattern.st]: Potentiating doesn't throw errors, but still doesn't work using `**`.

File: packages/AlgernonTests.package/ALGTestTypes.class/instance/testAssertEquationPattern.st [hunk 3]
```diff
@@ -3,5 +3,4 @@ testAssertEquationPattern
 
 	self assert: ['1+ 1 + 1' matchesRegex: ALGExpressionType equationPattern.].
 	self assert: ['-1 - 11' matchesRegex: ALGExpressionType equationPattern.].
-	self assert: ['2^ 3' matchesRegex: ALGExpressionType equationPattern.].
 	self assert: ['2/2' matchesRegex: ALGExpressionType equationPattern.].
```
Comment (yannik-dittmar) [inline packages/AlgernonTests.package/ALGTestTypes.class/instance/testAssertEquationPattern.st:6]: Maybe add another test for divide by zero?

**Review by yannik-dittmar — CHANGES_REQUESTED** [review]
Syntax error when placing a space between two numbers:
image

File: packages/Algernon.package/ALGExpressionType.class/class/equationPattern.st (hunk 2 above)
Comment (matveysiva) [reply packages/Algernon.package/ALGExpressionType.class/class/equationPattern.st]: We removed it intentionally because it would require a much more complex RegEx. For now we reduced equation solving to the basic operators only

**Review by yannik-dittmar — APPROVED** [review]
LGTM
